### PR TITLE
Add Marketplace readme

### PR DESCRIPTION
**Pull request by bgrgicak**
### Changes proposed in this Pull Request:

This PR contains a readme for the in-app Marketplace.
It's goal is to give an overview of the feature and help future developers get started.

The readme doesn't cover the helper API because it's part of a different plugin (WooCommerce), we should consider adding a readme in the future.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Read the readme
2. Check if anything is missing
3. Feel free to commit any changes to the readme

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+- **components**: Contains the React components used in the Marketplace page.
+- **contexts**: Contains the React contexts used in the Marketplace page.
+- **utils**: Contains the functionalities used in the Marketplace page to interact with APIs.
+- **stylesheets**: Contains the shared stylesheets used in the Marketplace page.
+- **assets**: Contains the images used in the Marketplace page.
+
+## Development
+
+This feature is part of WooCommerce Admin and uses the [same development environment.](/plugins/woocommerce-admin/README.md)
```
Comment (Dan-Q) [inline plugins/woocommerce-admin/client/marketplace/README.md:46]:
```suggestion
This feature is part of WooCommerce Admin and uses the [same development environment.](/plugins/woocommerce-admin/README.md)

```

Needs a trailing LF.

**Review by Dan-Q — APPROVED** [review]
Good explanation!

File: plugins/woocommerce-admin/client/marketplace/README.md (hunk 1 above)
Comment (Dan-Q) [reply plugins/woocommerce-admin/client/marketplace/README.md:46]: NM, you got it!

File: plugins/woocommerce-admin/client/marketplace/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+## Marketplace
+
+- **Discover**: This tab contains the featured extensions and themes.
+- **Browse**: This tab contains the list of all the extensions.
+- **Themes**: This tab contains the list of all the themes.
+- **Search**: This tab contains the search results.
+
+### Marketplace API
+
+The data for the Discover section is fetched from the `/wc/v3/marketplace/featured` endpoint.
```
Comment (mlaetitia) [inline plugins/woocommerce-admin/client/marketplace/README.md]: Isn't this also coming from wccom? It seems to call https://woocommerce.com/wp-json/wccom-extensions

File: plugins/woocommerce-admin/client/marketplace/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Marketplace
+
+This folder contains the components used in the Marketplace page.
```
Comment (mlaetitia) [inline plugins/woocommerce-admin/client/marketplace/README.md]: While we know and call this the marketplace, I don't recall it being written on the menu or pages. I'd put instead 
```suggestion
This folder contains the components used in the Marketplace page  (WooCommerce -> Extensions) .
```

File: plugins/woocommerce-admin/client/marketplace/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+The page contains two parts, the Woo.com marketplace and a list of products that the user purchased.
+
+## Marketplace
+
+- **Discover**: This tab contains the featured extensions and themes.
```
Comment (mlaetitia) [inline plugins/woocommerce-admin/client/marketplace/README.md]:
```suggestion
- **Discover**: This tab contains a cureated list of extensions and themes.
```

File: plugins/woocommerce-admin/client/marketplace/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+- **Search**: This tab contains the search results.
+
+### Marketplace API
+
+The data for the Discover section is fetched from the `/wc/v3/marketplace/featured` endpoint.
+Themes, extensions, and search results are fetched directly from Woo.com.
+
+## My Subscriptions
+
+This tab contains the list of all the extensions and themes that the user has purchased.
```
Comment (mlaetitia) [inline plugins/woocommerce-admin/client/marketplace/README.md]:
```suggestion
This tab contains the list of all the extensions and themes that the user has purchased from the Woo.com Marketplace
```

File: plugins/woocommerce-admin/client/marketplace/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+The list is fetched from Woo.com and allows shop owners to install, update, and connect products.
+If a subscription is expired, the user will be prompted to renew it.
+
```
Comment (mlaetitia) [inline plugins/woocommerce-admin/client/marketplace/README.md:27]:
```suggestion

The user needs to connect the site to their Woo.com account to be able to do any of the mentioned actions.

```

Comment (mlaetitia) [reply plugins/woocommerce-admin/client/marketplace/README.md:27]: User / shop owner / ...

**Review by mlaetitia — APPROVED** [review]
All good! Left a couple of suggestions, but feel free to ignore them if they aren't accurate or you don't seem them as improvements!

File: plugins/woocommerce-admin/client/marketplace/README.md (hunk 2 above)
Comment (raicem) [reply plugins/woocommerce-admin/client/marketplace/README.md]: Yes, ultimately the data is fetched from WCCOM. But we hit the store endpoint and it behaves as a proxy to cache and serve the content to the page.

File: plugins/woocommerce-admin/client/marketplace/README.md (hunk 6 above)
Comment (raicem) [reply plugins/woocommerce-admin/client/marketplace/README.md:27]: or a merchant?

Comment (mlaetitia) [reply plugins/woocommerce-admin/client/marketplace/README.md:27]: Yap - merchant seems more accurate !

**Review by andfinally — APPROVED** [review]
Thanks, this at least will help orient someone who is new to this code. I've pushed some suggested tweaks in [05de4e472a60511cfe48b859b16986c13ac5be27](https://github.com/woocommerce/woocommerce/pull/41600/commits/05de4e472a60511cfe48b859b16986c13ac5be27), addressing @mlaetitia and @raicem's comments.

**Pull request merged**